lib/targetctl.js

```diff
@@ -166,14 +166,17 @@ function onRequest(req, callback) {
         break;
 
       case 'stop-cpu-profiling':
-        rsp.profile = agent().metrics.stopCpuProfiling();
-        rsp.notify = {
-          wid: worker.id,
-          pid: worker.pid,
-          cmd: 'cpu-profiling',
-          isRunning: false,
-        };
-        break;
+        agent().metrics.stopCpuProfiling(function(profile) {
```

Comment:
The pattern used in the rest of this `switch` statement is to extract async responders in functions and do `return asyncResonder(req, res, callback);`.